### PR TITLE
feat: add temporary field to default error

### PR DESCRIPTION
**Pull request by zepatrik**
## Related Issue or Design Document

This feature allows to easily tell the client which errors it should retry and which need intervention.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: error_default.go [hunk 1]
```diff
@@ -64,6 +64,14 @@ type DefaultError struct {
 	// required: true
 	ErrorField string `json:"message"`
 
+	// Temporary marks an error as temporary.
+	//
+	// Temporary errors are usually caused by a temporary issue and should be
+	// retried by the client after a reasonable backoff.
+	//
+	// required: true
+	TemporaryField bool `json:"temporary"`
```
Comment (aeneasr) [inline error_default.go:73]: I'd inverse this and say "no_retry" because this error might not even make it to the caller if something else fails (e.g. internal cluster network error).

**Review by aeneasr — CHANGES_REQUESTED** [review]

File: error_default.go (hunk 1 above)
Comment (zepatrik) [reply error_default.go:73]: No retry feels very weird, as you have to say `NoRetry = false` to specify it is retryable. But you want the default to be retry right?

Comment (aeneasr) [reply error_default.go:73]: Yes, default should be retry for most 5xx errors (e.g. 502 bad gateway) and and some 4xx errors (e.g. 429 too many requests).

Usually, one would use the `Retry-After` to indicate when the retry can happen again. However, most of the infrastructure currently doesn't support setting this value correctly.

To disable retry explicitly, one could set `Retry-After` to a date far in the future, for example.

In summary, I don't think setting this as a JSON key is an appropriate fix as it is very application specific. We should probably work with default HTTP headers that are understood by most clients already.

For the console, we can narrow down the codes that need retry, e.g. 429, 503, 502, 500

**Comment by zepatrik** [comment]
OK, closing as this fix is too specific.
